flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/optimize/program/FlinkChangelogModeInferenceProgram.scala

```diff
@@ -687,9 +701,24 @@ class FlinkChangelogModeInferenceProgram extends FlinkOptimizeProgram[StreamOpti
     private def visitSink(
         sink: StreamPhysicalRel,
         sinkRequiredTraits: Seq[UpdateKindTrait]): Option[StreamPhysicalRel] = {
-      val children = sinkRequiredTraits.flatMap(t => visitChildren(sink, t))
+      var throwable: Throwable = null
+      val children = sinkRequiredTraits.flatMap { t =>
+        try {
+          visitChildren(sink, t)
```

Comment:
Can we change the return type of `visitChildren`? Instead of catch exception?

Reply:
The return type of `visitChildren` is already `Option`. The `visitChildren` should throw exception if anything is wrong, because the underlying exception should be known by the users in some cases (e.g. unsupport exception when changelog source emit UPDATE_AFTER without UPDATE_BEFORE). The fallback logic belongs to who has multiple required traits.